manifests/plugin.pp

```diff
@@ -15,4 +17,16 @@
     ensure => installed,
     notify => Class['foreman::service'],
   }
+
+  if $config {
+    file { $config_file:
+      ensure  => file,
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0644',
+      content => $config,
+      require => Package[$real_package],
+      notify  => Class['foreman::service'],
```

Comment:
`foreman::plugin <| |>` already notifies the service from init.pp.  (Actually, line 18 too.)